src/IO/Observer/TypeOfObservation.hpp

```diff
@@ -0,0 +1,23 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <iosfwd>
+
+namespace observers {
+/*!
+ * \ingroup ObserversGroup
+ * \brief Specifies the type of observation.
+ */
+enum class TypeOfObservation {
+  /// The sender will only perform reduction observations
+  Reduction,
+  /// The sender will only perform volume observations
+  Volume,
+  /// The sender will perform both reduction and volume observations
+  ReductionAndVolume
+};
+
+std::ostream& operator<<(std::ostream& os, const TypeOfObservation& t);
```

Comment:
noexcept

Reply:
I think in general we haven't be `noexcept`ing stream operators. Should we do that everywhere? I can add an issue to do that (as well as mark it here).

Reply:
Done and here is the issue https://github.com/sxs-collaboration/spectre/issues/930